### PR TITLE
Add Work Order Pages

### LINKED ISSUE #119: Add Work Order Pages

### PR DESCRIPTION
**Pull request by aggarwal-h**
Closes #119.

### CODE REVIEW

**Comment by aggarwal-h** [comment]
This PR now includes: 
- WorkOrder pages (list, update)
- Shop Profile page
- Invitation List page
- Updates to the Invite page (now supports inviting people!)
- Ability to select shops in Quote Request creation page
- New Create Shop Page

**Comment by aggarwal-h** [comment]
@tevisdoe Can you help me test this? In my previous comment above, I have included a list of things that have been updated in this PR. I won't be merging this until your PR is merged.

**Comment by tevisdoe** [comment]
@aggarwal-h I'm just gonna be posting as I go. First thing I noticed, this shop hours widget doesn't play nice with larger window sizes.

image

image

**Comment by tevisdoe** [comment]
Additionally, I'm getting this error when trying to create a shop. I think this has to do with what we spoke about on discord earlier today, though.

image

**Comment by aggarwal-h** [comment]
@tevisdoe Yes, this is the same error. I didn't add the fix because you already did in your PR. So it should be fixed when you merge your branch. If your branch is ready, it would be great it if you could merge it now so we can test them both at once?

**Comment by tevisdoe** [comment]
On my browser, this profile widget is always slightly off screen. I think it should not extend the window, but instead fit onto the screen.
image

**Comment by tevisdoe** [comment]
@aggarwal-h Well, my branch is based on your branch. Do you want me to merge it into yours? I can do that.

**Comment by aggarwal-h** [comment]
> @aggarwal-h Well, my branch is based on your branch. Do you want me to merge it into yours? I can do that.

Yeah, merge it in this.

**Comment by tevisdoe** [comment]
Okay, I have merged the two branches. I will begin testing again.

**Comment by tevisdoe** [comment]
Currently when you click the create button for quote requests you are not routed back to the quote request details page, even if they are created successfully. I don't know if that's an issue with this PR or if that was just there, but yeah.

File: src/next-app/pages/quote-request.tsx [hunk 1]
```diff
@@ -303,22 +334,22 @@ const QuoteRequestPage: NextPage = () => {
...
               }),
               headers: {
-                'Authorization': `JWT ${authData.access}`,
+                Authorization: `JWT ${authData.access}`,
                 'Content-Type': 'application/json; charset=UTF-8',
               },
             }).then((response) => {
               console.log(response);
               response.json().then((response) => {
                 console.log(response);
```
Comment (tevisdoe) [inline src/next-app/pages/quote-request.tsx:351]: In regards to my previous comment, I think we just have to add a router push here.

**Comment by tevisdoe** [comment]
Getting this as an error when attempting to send invite emails:

image

Looks like it's not properly authenticating with SMTP. Might be since I haven't set anything up on my end for this?

**Comment by aggarwal-h** [comment]
> Getting this as an error when attempting to send invite emails:
> 
> image
> 
> Looks like it's not properly authenticating with SMTP. Might be since I haven't set anything up on my end for this?

Yeah, you will need my environment variables for this. I have tested this and it works.

File: src/next-app/pages/quote-request.tsx (hunk 1 above)
Comment (aggarwal-h) [reply src/next-app/pages/quote-request.tsx:351]: Oh right, this was implemented by Gilbert, I only added the Select field. But I'll update this. Good catch!

**Comment by tevisdoe** [comment]
When trying to create an appointment (so I can create a work order), I'm getting this error.

image

I can tell it has something to do with the appointment slots. I assume I'm just missing something about creating appointments though.

**Comment by tevisdoe** [comment]
Okay, other than the stuff I've pointed out everything else works. Honestly, as long as that quote request routing issue is fixed I'm fine with merging this.

**Comment by aggarwal-h** [comment]
> When trying to create an appointment (so I can create a work order), I'm getting this error.
> 
> image
> 
> I can tell it has something to do with the appointment slots. I assume I'm just missing something about creating appointments though.

For this, it works a bit differently. You first send the duration of the appointment you want. You get sent back a list of appointment slots. Then you can include the slots you want when creating your appointment. I haven't documented this anywhere, but I'll document it after the demo.

**Comment by aggarwal-h** [comment]
> Okay, other than the stuff I've pointed out everything else works. Honestly, as long as that quote request routing issue is fixed I'm fine with merging this.

Yup, I'm working on a fix right now.

**Comment by aggarwal-h** [comment]
@tevisdoe I have pushed fixes for both Quote Request stuff, and the the Profile modal going out of page.

**Review by tevisdoe — APPROVED** [review]
Yup, working good now. I'm confident on approving this. Good work. You should take a break, you deserve it.

**Pull request merged**